WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -481,6 +485,42 @@ private extension NotificationsViewController {
 
 
 
+// MARK: - Unread notifications caching
+//
+private extension NotificationsViewController {
+    /// Updates the cached list of unread notifications, and optionally reloads the results controller.
+    ///
+    func refreshUnreadNotifications(reloadingResultsController: Bool = true) {
+        guard let notes = tableViewHandler.resultsController.fetchedObjects as? [Notification] else {
+            return
+        }
+
+        let previous = unreadNotificationIds
+
+        // This is additive because we don't want to remove anything
+        // from the list unless we explicitly call
+        // clearUnreadNotifications()
+        notes.lazy.filter({ !$0.read }).forEach { note in
+            unreadNotificationIds.insert(note.objectID)
+        }
+        if previous != unreadNotificationIds && reloadingResultsController {
+            reloadResultsController()
+        }
+    }
+
+    /// Empties the cached list of unread notifications.
+    ///
+    func clearUnreadNotifications() {
+        let shouldReload = unreadNotificationIds.isEmpty
+        unreadNotificationIds.removeAll()
+        if shouldReload {
```

Comment:
(Maybe i got this wrong!) shouldn't this condition be actualy `Unread Notification ID's` is **NOT** empty? (that would be the only scenario in which the cached row heights would need a cleanup: when something actually changed).

Apologies if i got this backwards!